compiler/semtypes.nim

```diff
@@ -857,9 +863,9 @@ proc liftParamType(c: PContext, procKind: TSymKind, genericParams: PNode,
     result = addImplicitGenericImpl(c, newTypeS(tyGenericParam, c), nil)
 
   of tyStatic:
-    # proc(a: expr{string}, b: expr{nkLambda})
-    # overload on compile time values and AST trees
-    if paramType.n != nil: return # this is a concrete type
+    if paramType.base.kind != tyNone and paramType.n != nil:
+      # this is a concrete type
```

Comment:
What is a "concrete" tyStatic parameter? Doesn't it always result in generic specialization?

Reply:
It should have said "this is a concrete value". I promise to fix it after the merge. I'm currently working in another branch.